qiskit/synthesis/two_qubit/sqisw_decompose.py

```diff
@@ -0,0 +1,211 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name, non-ascii-name
+
+"""Synthesis of two-qubit unitaries using at most 3 applications of the sqrt(iSWAP) gate."""
+
+import cmath
+from typing import Optional, List
+import numpy as np
+
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.quantum_info.operators import Operator
+from qiskit.quantum_info.synthesis.two_qubit_decompose import TwoQubitWeylDecomposition
```

Comment:
it's a bit confusing that this code in qiskit/sythesis whereas two_qubit_decompose is in qiskit/quantum_info/synthesis.
we should probably deprecate qiskit/quantum_info/synthesis and move all the code there to qiskit/sythesis.
(but this should not be part of this PR).

Reply:
yes that's my intent. to put the new stuff in qiskit/synthesis and then move over the old stuff too

Reply:
Doing that properly will also (hopefully) help us break a bunch of import cycles.